### PR TITLE
Piping self paced pl course offerings into build your own workshop topics dropdown

### PR DESCRIPTION
**Pull request by hannahbergam**
This PR populates the topics list for the 'Build Your Own Workshop' option. It sets up the backend controller action to bring in only the data we needed (this seemed preferable to reusing the curriculum quick assign options but having to filter out the long list of other course types). 

I added a test (and mocked the server response) to cover this new functionality, and also verified it works with our actual PL course list (see pic below).

Screenshot 2024-07-03 at 4 55 19 PM

## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-2020

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx [hunk 1]
```diff
@@ -1193,18 +1199,20 @@ export class WorkshopForm extends React.Component {
                       aria-labelledby="dropdownMenuButton"
                     >
                       <ul style={styles.listItems}>
-                        {ALL_PL_TOPICS.map(label => (
+                        {Object.values(ALL_PL_TOPICS).map(object => (
```
Comment (TurnerRiley) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx]: Nit: could `object` be renamed to `topic`?

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js [hunk 2]
```diff
@@ -560,6 +560,46 @@ describe('WorkshopForm test', () => {
...
+    const courseField = wrapper.find('#course').first();
+    courseField.simulate('change', {
+      target: {name: 'course', value: 'Build Your Own Workshop'},
+    });
+    expect(wrapper.find('#topics')).to.have.lengthOf(1);
+    wrapper.find('#dropdownMenuButton').first().simulate('click');
+    // A user can select either the label or checkbox, so we expect 2 for each here
+    expect(wrapper.find('#myPlTestTopic')).to.have.lengthOf(2);
+    expect(wrapper.find('#mySecondTopic')).to.have.lengthOf(2);
+  });
```
Comment (TurnerRiley) [inline apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js:601]: Great test with checking for both label and checkbox!

File: dashboard/app/controllers/course_offerings_controller.rb [hunk 3]
```diff
@@ -35,6 +35,12 @@ def quick_assign_course_offerings
     render :ok, json: offerings.to_json
   end
 
+  def self_paced_pl_course_offerings
+    return head :bad_request unless current_user
+    offerings = CourseOffering.assignable_course_offerings(current_user).select {|co| co.get_participant_audience == 'teacher' && co.instruction_type == 'self_paced'}
+    render :ok, json: offerings.map(&:summarize_self_paced_pl).to_json
```
Comment (TurnerRiley) [inline dashboard/app/controllers/course_offerings_controller.rb]: Nit: Not sure if this is even a case that could happen but wondering if there should be safe chaining after `offerings` in the case that the query returns no results

**Review by TurnerRiley — APPROVED** [review]
Looking great!

File: dashboard/app/controllers/course_offerings_controller.rb (hunk 3 above)
Comment (hannahbergam) [reply dashboard/app/controllers/course_offerings_controller.rb]: Yesss good call. Added!

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx (hunk 1 above)
Comment (hannahbergam) [reply apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx]: Good call- more clear. Updated!

**Pull request merged**